### DISCUSSION TITLE
Generation Directory

### DISCUSSION THREAD

**Discussion by Exerosis** [a-h/templ#518] [Q&A] [answered]
Can we have the generated code mapped to a separate directory? For example:
/pages/whatever.templ -> /generated/whatever_templ.go

**Comment by a-h** [comment]
This has been asked for before, but I've never really understood why anyone would want it other than for aesthetic reasons of not liking that the files are next to each other, or, IIRC to  avoid having to configure a hot reload tool.

In Go, the directory structure defines the package, and at the moment, the process for finding the matching Go file from a templ file is to change .templ to _templ.go. With a directory option, it would be a bit more complex, and also, any developer coming to a project couldn't rely on `templ generate` being the only command you need to use, you'd have to use the right set of flags along with it to set the output directory.

So, currently, my thinking is to not add the option, since it makes the code more complex to maintain, opens up a vector for bugs, and doesn't have any clear benefits.

If there's a killer use case, I'd always reconsider, but I'd rather spend my time on CSS handling, which I think could be improved in templ.

**Reply by Exerosis** [reply]
Ah yeah I guess without a unified build tool it would be hard to make portable. I don't think I have any killer use cases, both aesthetics (not clicking the generated file by mistake) and simplified hot reload were the driving force behind the thought. Thank you!

**Reply by a-h** [reply]
There's a much better hot reload feature coming in the next release. It uses an fsnotify based watcher, along with special code generation in dev mode that means you don't need to recompile/restart your app if you're only changing text/html in your templates.

Looking forward to releasing it, just not 100% finished with the testing, because I've made some big changes to how Go parsing in templates is done, and want to test the LSP a bit more.

**Reply by Exerosis** [reply]
Ah okay, right now we are using air for hot reload with pongo2. I briefly tested templ generate --watch and the performance wasn't quite so good for us. My dev env for this project is containerized a few layers deep so the resources available are more limited. I'm testing the air config in docs today, looking forward to improved hotreload!
Cheers, epic project thanks for your work on it!

**Comment by Perusae** [comment, marked as answer]
> Can we have the generated code mapped to a separate directory? For example: /pages/whatever.templ -> /generated/whatever_templ.go

That is really going to mess with Go, as your package names will all be in the same folder.

If the goal is to hide the generated files, if your using Visual Studio Code, go to Settings, Search "Hide" and enter the below pattern. 

Pattern:

     **/**/*templ.go

**Reply by Exerosis** [reply]
Excellent, I don't use VSC that much so I wasn't aware this was possible, this solves it for me.

**Reply by joerdav** [reply]
And the jetbrains plugin does this automatically!

**Comment by a-h** [comment]
In the templ channel in the Gopher's Slack, Ali Yeysides said you can also use a vim command to hide the generated files:

```lua
function HideFiles()
  vim.cmd [[let g:netrw_list_hide.=',.*_templ.go']]
end

HideFiles()
```
